### DISCUSSION TITLE
rust or zig ?

### DISCUSSION THREAD

**Discussion by baxiry** [Amjad50/Emerald#43] [General]
Hello Amjad. You are doing a very wonderful job.
I wonder what you think about the Zig language?
Personally, I think it is the best in your field of interest: kernels and operating systems.
Have you studied or researched this topic?
What were your conclusions?
Thanks

**Comment by Amjad50** [comment]
I didn't try Zig before, probably it would work with kernels, as its very similar to C.

The language can have small effect on the lower kernel parts since we are using unsafe Rust in there. So both would be similar in that place.
But after that, I think Rust gives us the safety we need to perform further tasks safely.

But, I'm not planning to add Zig anytime soon anyway, its best to stay with 1 compilation tools and easier to maintain

**Comment by baxiry** [comment]
I read some articles that said that Zig is better for low-level programming where unsafe is common. 
I would love to know your opinion.
Anyway, keep up the good work. 
Thank you

**Comment by Amjad50** [comment]
Not sure which is best, since I haven't used Zig personally.
There are some projects in Zig related to this, example: https://github.com/ZystemOS/pluto.

In general, Rust is designed to also support embedded systems and low level systems that are running without OS which is what this kernel kinda is, since it's the OS. That's why it's working quite well.

Both are good and usable here, there is not only 1 way.

**Comment by baxiry** [comment]
@Amjad50 Thanks! <3
